### PR TITLE
chore: Address comments in the #2874

### PR DESCRIPTION
**Pull request by ssheng**
## What does this PR address?

Address comments in https://github.com/bentoml/BentoML/pull/2874.

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

Feel free to tag members/contributors who can help review your PR.

### CODE REVIEW

File: bentoml/_internal/io_descriptors/json.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
     )
 
 
-JSONType = t.Union[str, t.Dict[str, t.Any], "pydantic.BaseModel"]
+JSONType = t.Union[str, t.Dict[str, t.Any], "pydantic.BaseModel", None]
```
Comment (aarnphm) [inline bentoml/_internal/io_descriptors/json.py:41]:
```suggestion
JSONType = t.Optional[str, t.Dict[str, t.Any], "pydantic.BaseModel"]
```

**Review by aarnphm — DISMISSED** [review]

File: bentoml/_internal/io_descriptors/json.py (hunk 1 above)
Comment (ssheng) [reply bentoml/_internal/io_descriptors/json.py]: Optional seems to take only one input.
```
TypeError: Optional[t] requires a single type. Got (<class 'str'>, typing.Dict[str, typing.Any], 'pydantic.BaseModel').
```

Comment (aarnphm) [reply bentoml/_internal/io_descriptors/json.py:41]: Oh my bad, it has to be `t.Optional[t.Union[...]]`

Comment (aarnphm) [reply bentoml/_internal/io_descriptors/json.py:41]: but `t.Union[.., None]` works as well.

**Review by aarnphm — APPROVED** [review]

**Pull request merged**